src/linear_algebra/tensor_algebra/to_tensor_power.lean

```diff
@@ -0,0 +1,178 @@
+/-
+Copyright (c) 2021 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import linear_algebra.tensor_algebra.basic
+import linear_algebra.tensor_power
+/-!
+# Tensor algebras as direct sums of tensor powers
+
+In this file we show that `tensor_algebra R M` is isomorphic to a direct sum of tensor powers, as
+`tensor_algebra.equiv_direct_sum`.
+-/
+open_locale direct_sum tensor_product
+
+variables {R M : Type*} [comm_semiring R] [add_comm_monoid M] [module R M]
+
+namespace tensor_power
+
+/-- The canonical embedding from a tensor power to the tensor algebra -/
+def to_tensor_algebra {n} : ⨂[R]^n M →ₗ[R] tensor_algebra R M :=
+pi_tensor_product.lift (tensor_algebra.tprod R M n)
+
+@[simp]
+lemma to_tensor_algebra_tprod {n} (x : fin n → M) :
+  tensor_power.to_tensor_algebra (pi_tensor_product.tprod R x) = tensor_algebra.tprod R M n x :=
+pi_tensor_product.lift.tprod _
+
+@[simp]
+lemma to_tensor_algebra_ghas_one :
```

Comment:
Mentioning the name of the instance in the lemma is surprising to me. Why so?

Reply:
This is about `ghas_one.one`, not `has_one.one`; we don't have a convention in place for lemmas about the former.

Reply:
I would have suggested `gone` 💨